src/data/list/duplicate.lean

```diff
@@ -0,0 +1,234 @@
+/-
+Copyright (c) 2021 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky, Chris Hughes
+-/
+import data.list.nodup
+import data.fin
+
+/-!
+# List duplicates
+
+## Main definitions
+
+* `list.duplicate x l : Prop` is an inductive property that holds when `x` is a duplicate in `l`
+
+-/
+
+variable {α : Type*}
+
+namespace list
+
+/-- Property that an element `x : α` of `l : list α` can be found in the list more than once. -/
+inductive duplicate (x : α) : list α → Prop
+| cons_mem {l : list α} : x ∈ l → duplicate (x :: l)
+| cons_duplicate {y : α} {l : list α} : duplicate l → duplicate (y :: l)
+
+local infix ` ∈+ `:50 := list.duplicate
+
+variables {l : list α} {x : α}
+
+lemma mem.duplicate_cons_self (h : x ∈ l) : x ∈+ x :: l := duplicate.cons_mem h
+
+lemma duplicate.duplicate_cons (h : x ∈+ l) (y : α) : x ∈+ y :: l := duplicate.cons_duplicate h
```

Comment:
won't dot notation work with `cons_duplicate`? why bother duplicating it? (heh)

Reply:
It does work with `cons_duplicate`, but that has `{y : α}` as an implicit arg, while this has it as an explicit arg.